tests/mypy/outputs/plugin-fail.txt

```diff
@@ -8,6 +8,7 @@
 58: error: Invalid value for "Config.orm_mode"  [pydantic-config]
 63: error: Invalid value for "Config.orm_mode"  [pydantic-config]
 74: error: Incompatible types in assignment (expression has type "ellipsis", variable has type "int")  [assignment]
+83: error: Incompatible types in assignment (expression has type "None", variable has type "int")  [assignment]
```

Comment:
Thank you - I am not sure though that I correctly understood why this should not fail.

In both strict and non-strict mode, shouldn't a code like this fail?
```
x: int = None
```

That is basically what it is being done in that instance; we are assigning `None` to a variable that has `int` as a type. I admit I am very noob when it comes to strict/non-strict mypy differences, but I thought this kind of check should fail in both modes.

Following your comment https://github.com/samuelcolvin/pydantic/pull/3430#discussion_r762456160 I think it would be even better though to have a message like:
```Incompatible type in `default`(...) ```

It would make way more clear what is failing to the reader; to do so, my best bet is to use the `is_overlapping_type` API from `mypy`... I'll give it a try, thank you for the feedback!

Reply:
Perhaps, you can look at this example in docs: https://pydantic-docs.helpmanual.io/usage/models/#recursive-models

Previously, you could find same pattern in the first example: https://github.com/samuelcolvin/pydantic/issues/1072

You can think of declaring pydantic fields as declaring function arguments, where `a: int = None` is allowed.

However, it also worth looking on the reason why the original example was changed. Maybe pydantic is moving towards more conventional typing model, in which case such change in the plugin should be welcomed.

Nevertheless IMO it should be introduced in separate PR.

Reply:
Thinking about this more, I came to conclusion that `a: int = Field(None)` should produce the same result as `a: int = None` would. 

I didn't find such assignment in the test file and assumed it doesn't produce error. If it does, then field default should indeed raise error as well.

TL;DR: let's add `a: int = None` and match it's behavior with Field \o/

Reply:
Thank you again for the feedback, super helpful!

I will add the example you specified then 🚀

Reply:
Added the test you suggest here: https://github.com/samuelcolvin/pydantic/pull/3430/commits/0380b43c7d2c87d6fde3379ade36cc68ba9563e3 Thank you!

Reply:
I agree with you both.

Pydantic will be changing the way it deals with optional in v2, see #2353.

But for now I agree with this.